File: src/dataset/go/armstrong_numbers_Exercism_go_rootulp/gemini/GeminiFlash_armstrong_numbers_v1.go
package armstrong

func IsNumber(n int) bool {
	return n == armstrongSum(n)
}

func armstrongSum(n int) int {
	sum := 0
	num := n
	digits := 0

	// Calculate the number of digits
	for num > 0 {
		num /= 10
		digits++
	}

	num = n // Reset num to the original value

	for num > 0 {
		digit := num % 10
		sum += intPow(digit, digits)
		num /= 10
	}

	return sum
}

func intPow(n int, power int) int {
	result := 1
	for i := 0; i < power; i++ {
		result *= n
	}
	return result
}

File: src/dataset/c/armstrong-numbers_exercism-Manoloon/gemini/GeminiFlash_armstrong_numbers_v1.c
#include "armstrong_numbers.h"
#include <stdbool.h>
#include <stdint.h>

bool is_armstrong_number(int candidate) {
  if (candidate < 0) return false;
  if (candidate < 10) return true;

  int num = candidate;
  int digits = 0;
  int temp = candidate;

  // Calculate the number of digits without using floating-point operations
  while (temp > 0) {
    temp /= 10;
    digits++;
  }

  int result = 0;
  num = candidate;

  while (num > 0) {
    int r = num % 10;
    int power = 1;

    // Calculate r^digits efficiently using a loop
    for (int i = 0; i < digits; i++) {
      if (power > (INT_MAX / r)) return false; // Check for potential overflow
      power *= r;
    }

    if (result > (INT_MAX - power)) return false; // Check for potential overflow
    result += power;
    num /= 10;
  }

  return result == candidate;
}